packages/kbn-shared-ux-components/src/page_template/solution_nav/solution_nav.tsx

```diff
@@ -67,68 +93,82 @@ export const KibanaPageTemplateSolutionNav: FunctionComponent<
   };
 
   const isHidden = isLargerBreakpoint && !isOpenOnDesktop;
+  const isCustomSideNav = !!children;
+
+  const sideNavClasses = classNames('kbnPageTemplateSolutionNav', {
+    'kbnPageTemplateSolutionNav--hidden': isHidden,
+  });
 
   /**
    * Create the avatar
    */
-  const solutionAvatar = icon ? (
-    <KibanaSolutionAvatar
-      className="kbnPageTemplateSolutionNavAvatar"
-      iconType={icon}
-      name={name}
-    />
-  ) : null;
+  const solutionAvatar = useMemo(() => {
+    if (!icon) {
+      return;
+    }
+    return (
+      <KibanaSolutionAvatar
+        className="kbnPageTemplateSolutionNav__avatar"
+        iconType={icon}
+        name={name}
+      />
+    );
+  }, [name, icon]);
 
   /**
    * Create the titles
    */
-  const titleText = (
-    <>
-      {solutionAvatar}
-      <strong>{name}</strong>
-    </>
-  );
-  const mobileTitleText = (
-    <FormattedMessage
-      id="sharedUXComponents.solutionNav.mobileTitleText"
-      defaultMessage="{solutionName} Menu"
-      values={{ solutionName: name || 'Navigation' }}
-    />
+  const headingID = headingProps?.id || generateId('heading');
+  const HeadingElement = headingProps?.element || 'h2';
+  const titleText = useMemo(
+    () => (
+      <EuiTitle size="xs" id={headingID}>
+        <HeadingElement>
+          {solutionAvatar}
+          <strong>{name}</strong>
+        </HeadingElement>
+      </EuiTitle>
+    ),
+    [name, solutionAvatar, headingID, HeadingElement]
   );
 
   /**
-   * Create the side nav component
+   * Create the side nav content
    */
-
-  const sideNav = () => {
+  const sideNavContent = useMemo(() => {
+    if (isCustomSideNav) {
+      return children;
+    }
     if (!items) {
       return null;
     }
-    const sideNavClasses = classNames('kbnPageTemplateSolutionNav', {
-      'kbnPageTemplateSolutionNav--hidden': isHidden,
-    });
     return (
       <EuiSideNav
+        aria-labelledby={headingID}
         aria-hidden={isHidden}
-        className={sideNavClasses}
-        heading={titleText}
-        mobileTitle={
-          <>
-            {solutionAvatar}
-            {mobileTitleText}
-          </>
-        }
-        toggleOpenOnMobile={toggleOpenOnMobile}
-        isOpenOnMobile={isSideNavOpenOnMobile}
         items={setTabIndex(items, isHidden)}
+        mobileBreakpoints={[]} // prevent EuiSideNav to apply mobile version, already implemented here
         {...rest}
       />
     );
-  };
+  }, [children, headingID, isCustomSideNav, isHidden, items, rest]);
 
   return (
     <>
-      {isSmallerBreakpoint && sideNav()}
+      {isSmallerBreakpoint && (
+        <EuiCollapsibleNavGroup
```

Comment:
@semd I've simplified this new mobile view to use this component instead. We don't need to replicate EuiSideNav's full customization since we actually want to control Kibana's usages to be consistent especially with accessibility. This does look slightly different from the original, but that's ok.